scripts/tests/test_zkai_jstprove_statement_envelope_benchmark.py

```
@@ -0,0 +1,232 @@
+        envelope["artifacts"]["proof_path"] = "../proof.msgpack"
+
+        with self.assertRaisesRegex(BENCH.JstproveEnvelopeError, "not an allowed JSTprove artifact"):
+            BENCH.verify_statement_envelope(envelope, external_verify=fake_external_verify)
+
+    def test_jstprove_verify_times_out_fail_closed(self) -> None:
+        with mock.patch.object(BENCH.shutil, "which", return_value="/usr/bin/jstprove-remainder"):
+            with mock.patch.object(
+                BENCH.subprocess,
+                "run",
+                side_effect=BENCH.subprocess.TimeoutExpired(cmd=["jstprove-remainder"], timeout=180),
+            ):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)